File: share-code/bai-tap-lap-trinh/python/loop/52.py
yourNumber = int(input("Nhập số nguyên: "))

num_str = str(yourNumber)

# Cách 1:
digit_sum = 0

for digit in num_str:
    digit_sum += int(digit)

# Cách 2:
# digit_sum = sum(int(digit) for digit in num_str)

print(f"Total: {len(num_str)} digits")
print(f"Sum: {' + '.join(num_str)} = {digit_sum}")

File: share-code/bai-tap-lap-trinh/cpp/52.cpp
#include<iostream>
using namespace std;
int main()
{
    // in số chữ số của 1 số nguyên dương
    // tính tổng của các chữ số
    int yourNumber      = 12;
    int sum             = 0;
    int count           = 0;

    while (yourNumber > 0)
    {
        int rightNumber     = yourNumber % 10;
        sum                 += rightNumber;
        yourNumber          = yourNumber / 10;
        count++;
    }

    cout << "sum: " << sum << endl;
    cout << "count: " << count << endl;
}